src/components/core/TextArea/TextArea.tsx

```diff
@@ -30,6 +30,47 @@ export const TextArea = forwardRef<HTMLTextAreaElement, TextAreaProps>(
     },
     ref
   ) => {
+    // If the user is pasting a link, automatically place it in a Markdown link
+    const handleKeyDown = useCallback(
+      async (event) => {
+        if (event.key === 'v' && event.metaKey) {
+          // Wrap in a try/catch because navigator.clipboard.readText() is not supported in Firefox. If it fails, pasting behavior will be default
+          try {
+            // FIXME: Handle Typescript error from forwardRef
+            if (ref.current) {
+              const textArea = ref.current;
+              const [selectionStart, selectionEnd] = [
+                textArea.selectionStart,
+                textArea.selectionEnd,
+              ];
+              const selectedText = textArea.value.substring(selectionStart, selectionEnd);
+              const pastedText = await navigator.clipboard.readText();
+              if (pastedText.startsWith('http')) {
+                event.preventDefault();
+
+                const textBeforeLink = textArea.value.substring(0, selectionStart);
+                const textAfterLink = textArea.value.substring(
+                  selectionStart + pastedText.length,
+                  selectionEnd + pastedText.length
+                );
+                const newValue = `${textBeforeLink}[${selectedText}](${pastedText})${textAfterLink}`;
+
+                const newSelectionStart = selectionStart + 1;
+                const newSelectionEnd = selectionStart + selectedText.length + 1;
```

Comment:
`selectionStart + 1` accounts for the new starting bracket `[`, and `selectionStart + selectedText.length + 1` accounts for the starting bracket `[` and `<highlighted text>`. So when the user pastes, their new "selection" (e.g. the new highlight) is `[<one character past old selection string>, <end of old selection string>]`.

Reply:
woah, this is sick.

let's leave this comment in the code as a comment